File: 0125-valid-palindrome/0125-valid-palindrome.py
class Solution:
    def isPalindrome(self, s: str) -> bool:
        p2 = len(s) - 1
        p1 = 0
        while(p1<p2):
            while((s[p1].isalnum()) == False and p1<p2):
                p1 += 1
            while((s[p2].isalnum()) == False and p1<p2):
                p2 -= 1
            if(s[p1].lower()!=s[p2].lower()):
                return False
            p1 += 1
            p2 -= 1
        return True
    '''
    race a car
    |        |

    '''

File: 0125-valid-palindrome/0125-valid-palindrome.cpp
class Solution {
public:
    bool isPalindrome(string s) {
        vector<char> new_s;
        for(auto &l:s){
            l = tolower(l);
            if(l>=97 && l<=122 || l>=48 && l<=57)
                new_s.push_back(l);
        } 
        int n,p2=new_s.size();
        n = p2/2;
        if(p2%2!=0)
            n++;
        p2--;
        for(int p1=0;p1<n;p1++){
            if(new_s[p1]!=new_s[p2-p1])
                return false;
        }
        
        return true;
    }
};